src/mame/drivers/mekd1.cpp

```diff
@@ -0,0 +1,462 @@
+
+	virtual void device_timer(emu_timer &timer, device_timer_id id, int param, void *ptr) override;
+
+	emu_timer *m_bit_rate_timer;
+	emu_timer *m_bit_rate_half_timer;
+
+	bool m_bit_rate_out;
+	bool m_bit_rate_half_out;
+	bool m_bit_rate_reset;
+	bool m_bit_rate_select;
+};
+
+void mekd1_state::mem_map(address_map &map)
+{
+	map.unmap_value_high();
+	map(0x0000, 0x027f).mirror(0x7c00).ram();
+
+	// The emulated addressing for these is not exact. The hardware uses
+	// A3, A4, and A5 to select between these respectively. For example
+	// the first PIA is selected when A3 is high, but it ignores A4 and
+	// A5, etc. So it is possible that more than one are selected at once
+	// which is not emulated here. Here it is assumed that A3, A4, and A5
+	// are completely decoded and that only one device is selected at a
+	// time.
+	map(0x8004, 0x8007).mirror(0x5fe0).rw(m_pia0, FUNC(pia6821_device::read), FUNC(pia6821_device::write));
+	map(0x8008, 0x800b).mirror(0x5fe0).rw(m_pia1, FUNC(pia6821_device::read), FUNC(pia6821_device::write));
+	map(0x8010, 0x8011).mirror(0x5fe2).rw(m_acia, FUNC(acia6850_device::read), FUNC(acia6850_device::write));
+
+	map(0xa000, 0xa07f).mirror(0x0f80).ram();
+
+	// Although the ROM is 1k, MIKBUG uses only the first 512 bytes and
+	// when using MIKBUG the A9 ROM input is connected to 0V.
+	map(0xe000, 0xe1ff).mirror(0x1e00).rom().region("mcm6830l7", 0);
+}
+
+static INPUT_PORTS_START( mekd1 )
+
+	PORT_START("baud_rate")
+	PORT_CONFNAME(0x3fff, 416, "RS232 Baud Rate")
+	PORT_CONFSETTING(9091, "110")
+	PORT_CONFSETTING(3333, "300")
+	PORT_CONFSETTING(1667, "600")
+	PORT_CONFSETTING( 833, "1200")
+	PORT_CONFSETTING( 416, "2400")
+	PORT_CONFSETTING( 208, "4800")
+	PORT_CONFSETTING( 139, "7200")
+	PORT_CONFSETTING( 104, "9600")
+
+	PORT_START("stop_bits")
+	PORT_CONFNAME(0x01, 0, "Stop bits")
+	PORT_CONFSETTING(0x00, "1")
+	PORT_CONFSETTING(0x01, "2")
+
+	PORT_START("acia_baud_rate")
+	PORT_CONFNAME(0xf, 1, "ACIA Baud Rate")
+	PORT_CONFSETTING(13, "110")
+	PORT_CONFSETTING(11, "150")
+	PORT_CONFSETTING(9, "300")
+	PORT_CONFSETTING(8, "600")
+	PORT_CONFSETTING(7, "1200")
+	PORT_CONFSETTING(5, "2400")
+	PORT_CONFSETTING(4, "3600")
+	PORT_CONFSETTING(2, "4800")
+	PORT_CONFSETTING(1, "9600")
+
+INPUT_PORTS_END
+
+
+
+READ8_MEMBER(mekd1_state::pia0_pa_r)
+{
+	return m_rs232->rxd_r() << 7;
+}
+
+READ8_MEMBER(mekd1_state::pia0_pb_r)
+{
+	bool timer_out;
+	uint8_t stop_bits = m_stop_bits->read();
+
+	if (m_bit_rate_select)
+		timer_out = m_bit_rate_out;
+	else
+		timer_out = m_bit_rate_half_out;
+
+	return (timer_out << 7) | (stop_bits << 6);
+}
+
+WRITE8_MEMBER(mekd1_state::pia0_pa_w)
+{
+	m_rs232->write_txd(BIT(data, 0));
+}
+
+WRITE8_MEMBER(mekd1_state::pia0_pb_w)
+{
+	m_bit_rate_select = BIT(data, 2);
+
+	if (BIT(data, 0) == 1)
+	{
+		// Reset
+		m_bit_rate_timer->reset();
+		m_bit_rate_half_timer->reset();
+		m_bit_rate_out = 0;
+		m_bit_rate_half_out = 0;
+		m_bit_rate_reset = 1;
+		return;
+	}
+
+	if (m_bit_rate_reset)
+	{
```

Comment:
Please be consistent with brace style - this is Allman, while you have K&R in other places.